### PR TITLE
Feature/implementation build messages

### PR DESCRIPTION
**Pull request by bashnia007**

### CODE REVIEW

File: CommonLibrary/Enumerations/GameMessageType.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@ public enum GameMessageType
         ActionMessage,
         BuildMessage,
         InitializeMessage,
-        UpdateTableMessage
+        UpdateTableMessage,
+        DirectMessage,
```
Comment (Darina1801) [inline CommonLibrary/Enumerations/GameMessageType.cs]: Make it in alphabetic order, please

File: CommonLibrary/Message/BuildMessage.cs [hunk 2]
```diff
@@ -1,11 +1,22 @@
-﻿namespace CommonLibrary.Message
+﻿using System;
```
Comment (Darina1801) [inline CommonLibrary/Message/BuildMessage.cs]: Make it in alphabetic order, please

File: Saboteur/ViewModel/MainViewModel.cs [hunk 3]
```diff
@@ -220,13 +215,44 @@ private void HandleTextMessage(TextMessage message)
...
+
+        private void HandleDirectMessage(DirectMessage message)
+        {
+            _isMyTurn = message.IsMyTurn;
+        }
+
+        private void PrepareMap()
+        {
+            Map = new ObservableCollection<ObservableCollection<RouteCard>>();
+            for (int i = 0; i < 6; i++)
```
Comment (Darina1801) [inline Saboteur/ViewModel/MainViewModel.cs]: Change int name

File: Saboteur/ViewModel/MainViewModel.cs [hunk 4]
```diff
@@ -220,13 +215,44 @@ private void HandleTextMessage(TextMessage message)
...
+            {
+                var row = new List<RouteCard>();
+                for (int j = 0; j < 9; j++)
```
Comment (Darina1801) [inline Saboteur/ViewModel/MainViewModel.cs]: Change int name

File: Server/ServerObject.cs [hunk 5]
```diff
@@ -7,6 +7,7 @@
 using System.Net.Sockets;
 using System.Runtime.Serialization.Formatters.Binary;
 using System.Threading;
+using CommonLibrary.Enumerations;
```
Comment (Darina1801) [inline Server/ServerObject.cs]: Change order of usings

File: CommonLibrary/Enumerations/GameMessageType.cs (hunk 1 above)
Comment (bashnia007) [reply CommonLibrary/Enumerations/GameMessageType.cs]: done

File: CommonLibrary/Message/BuildMessage.cs (hunk 2 above)
Comment (bashnia007) [reply CommonLibrary/Message/BuildMessage.cs]: done

File: CommonLibrary/Message/BuildMessage.cs [hunk 6]
```diff
@@ -1,11 +1,22 @@
...
+            MessageType = GameMessageType.BuildMessage;
+        }
         // ID карты, которую игрок хочет построить
         public int CardId { get; set; }
+        public RouteCard RouteCard { get; set; }
         // координаты, где игрок хочет построить карту тунеля
-        public int X { get; set; }
-        public int Y { get; set; }
+        public Coordinates Coordinates { get; set; }
+
```
[the remark this answers is not in the document]
Comment (bashnia007) [reply CommonLibrary/Message/BuildMessage.cs]: done

File: CommonLibrary/Message/DirectMessage.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿using CommonLibrary.Enumerations;
+using System;
+
+namespace CommonLibrary.Message
+{
+    [Serializable]
+    public class DirectMessage : Message
+    {
+        public DirectMessage()
```
[the remark this answers is not in the document]
Comment (bashnia007) [reply CommonLibrary/Message/DirectMessage.cs]: It is a class, not method=) Done

File: Saboteur/ViewModel/MainViewModel.cs [hunk 8]
```diff
@@ -97,12 +79,19 @@ public MainViewModel(string login)
...
         private void ExecuteBuildTunnelCommand(object obj)
         {
-
+            var mapItem = (RouteCard) obj;
+            _client.SendMessage(new BuildMessage
+            {
+                Coordinates = mapItem.Coordinates,
+                SenderId = CurrentPlayer.Id,
+                CardId = SelectedCard.Id,
+                RouteCard = SelectedCard as RouteCard
```
[the remark this answers is not in the document]
Comment (bashnia007) [reply Saboteur/ViewModel/MainViewModel.cs]: Will see

File: Saboteur/ViewModel/MainViewModel.cs (hunk 3 above)
Comment (bashnia007) [reply Saboteur/ViewModel/MainViewModel.cs]: done

File: Saboteur/ViewModel/MainViewModel.cs (hunk 4 above)
Comment (bashnia007) [reply Saboteur/ViewModel/MainViewModel.cs]: done

File: Server/ServerObject.cs (hunk 5 above)
Comment (bashnia007) [reply Server/ServerObject.cs]: done

**Comment by bashnia007** [comment]
#42

**Comment by bashnia007** [comment]
#36

**Pull request merged**